python/kserve/kserve/model_server.py

```diff
@@ -226,7 +227,7 @@ def start(self, models: Union[List[Model], Dict[str, Deployment]]) -> None:
         """
         if isinstance(models, list):
             for model in models:
-                if isinstance(model, Model):
+                if isinstance(model, Union[Model, OpenAIModel]):
```

Comment:
I wanted to make this change separately and add unit tests to it. However the model registry logic is too tightly couple with starting the server processes. Hence I couldn't unit test them. Since this change turned out to be small and related, I am appending it to this PR.

Reply:
cc @cmaddalozzo let me know ifyou have feedback on https://github.com/kserve/kserve/pull/3590/commits/b16fc8ddef46ca496343d5a45a562d07834b1ecd

Reply:
Let's make `isinstance(model, Union[Model, OpenAIModel])` a function so we do not need to change so many places next time.

Reply:
I made it a type: `BaseModelType = Union[Model, OpenAIModel]` Next time we can only update this type.

Reply:
Let me know if this is not addressing the concern.